Makefile

```diff
@@ -21,7 +21,10 @@ build: build-web build-go
 build-go:
 	go build $(GO_BUILD_TAGS) ./...
 
-build-web:
+build-web: web/runtime-shared/static/control.js
+
+# this assumes that control.js is always updated when this target runs
+web/runtime-shared/static/control.js: web/frontend/src/*/* web/frontend/src/*/*/* web/frontend/src/*.* web/frontend/scripts/* web/frontend/*.*
```

Comment:
I'm concerned that this may cause confusion in the future if a new folder is added, and its not clear why changes aren't being picked up. But we cannot just include the base `web/frontend` folder because `static` is a subdirectory. So this is fine.